src/commands/navigation.rs

```diff
@@ -192,11 +198,33 @@ pub fn checkout(effects: &Effects, git_run_info: &GitRunInfo) -> eyre::Result<is
     let graph = make_smartlog_graph(effects, &repo, &dag, &event_replayer, event_cursor, true)?;
 
     match prompt_select_commit(graph.get_commits(), None)? {
-        Some(oid) => {
-            let result = git_run_info.run(effects, None, &["checkout", &oid.to_string()])?;
-            smartlog(effects, &Default::default())?;
-            Ok(result)
-        }
+        Some(oid) => checkout_commit(effects, git_run_info, oid),
         None => Ok(1),
     }
 }
+
+fn checkout_commit<T>(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    target: T,
+) -> eyre::Result<isize>
+where
+    T: ToString,
+{
+    let result = git_run_info.run(effects, None, &["checkout", &target.to_string()])?;
+    smartlog(effects, &Default::default())?;
+    if result != 0 {
+        writeln!(
+            effects.get_output_stream(),
+            "{}",
+            printable_styled_string(
+                effects.get_glyphs(),
+                StyledString::styled(
+                    format!("Failed to checkout commit {}.", target.to_string(),),
```

Comment:
This should read `Failed to check out commit: {}`:

- The verb form is "check out", and the noun form is "checkout". It's unfortunate that the Git command is called `checkout`; even in its own documentation, it uses the phrase "check out" for the verb.
- Likewise, rename this function to `check_out_commit`.
- In the event that the user wants to copy-and-paste the commit identifier, it's best to not include the `.` as part of the word, so that their selection doesn't accidentally copy the extra `.`.

Can you also update `execute::check_out_updated_head` to use this function?

Reply:
This function should also take an explicit `Option<EventTransactionId>` and forward it to `run`. If the checkout is part of a larger overall operation (such as a rebase/move/undo),  then it should be associated with the overall operation, so that the operation can be rendered/undone as a single unit.

Reply:
👍 
(Good thought about removing the ".")